ansible/library/conn_graph_facts.py

```diff
@@ -205,6 +206,37 @@ def _port_alias_to_name(self, device, port):
     def _get_sorted_port_name_list(self, hwsku):
         return natsorted(self._get_port_alias_to_name_map(hwsku).values())
 
+    def _get_port_name_to_alias_map(self, hwsku):
+        """
+        Retrive port name to alias map for specific hwsku.
+        """
+        if hwsku in self._cache_port_name_to_alias:
+            return self._cache_port_name_to_alias[hwsku]
+        port_alias_to_name_map = self._get_port_alias_to_name_map(hwsku)
+        port_name_to_alias_map = dict([(name, alias) for alias, name in port_alias_to_name_map.items()])
+        self._cache_port_name_to_alias[hwsku] = port_name_to_alias_map
+        return port_name_to_alias_map
+
+    def _get_port_name_set(self, device_hostname):
+        """
+        Retrive port name set of a specific hwsku.
+        """
+        hwsku = self.graph_facts["devices"][device_hostname]['HwSku']
+        return set(self._get_port_name_to_alias_map(hwsku).keys())
+
+    def _get_port_alias_set(self, device_hostname):
+        """
+        Retrive port alias set of a specific hwsku.
+        """
+        hwsku = self.graph_facts["devices"][device_hostname]['HwSku']
+        return set(self._get_port_alias_to_name_map(hwsku).keys())
+
+    def sorted_dict(self, myDict):
+        myKeys = list(myDict.keys())
+        myKeys.sort()
+        rst_dict = {i: myDict[i] for i in myKeys}
+        return rst_dict
```

Comment:
Two comments:
1. The `rst_dict` would still be unsorted. In python, sequence of keys cannot be guaranteed. No matter how the dict was built.
2. This function is not used anywhere, remove it?

Reply:
Thanks, will remove it.